katello/rubygem-katello/rubygem-katello.spec

```diff
@@ -0,0 +1,1788 @@
+BuildRequires: foreman-plugin >= 1.11.0
+BuildRequires: %{?scl_prefix}rubygem(foreman_docker) >= 0.2.0
+BuildRequires: %{?scl_prefix}rubygem(angular-rails-templates) >= 0.0.4
+BuildRequires: %{?scl_prefix}rubygem(bastion) >= 6.0.0
+BuildRequires: %{?scl_prefix}rubygem(bastion) < 7.0.0
+BuildRequires: %{?scl_prefix}rubygem(foreman-tasks) >= 0.8.0
+BuildRequires: %{?scl_prefix}rubygem(foreman-tasks) < 1.0.0
+BuildRequires: %{?scl_prefix}rubygem(gettext_i18n_rails)
+BuildRequires: %{?scl_prefix}rubygem(apipie-rails) >= 0.1.1
+BuildRequires: %{?scl_prefix}rubygem(runcible) >= 2.0.0
+BuildRequires: %{?scl_prefix}rubygem(anemone)
+BuildRequires: %{?scl_prefix}rubygem(deface) >= 1.0.0
+BuildRequires: %{?scl_prefix}rubygem(deface) < 2.0.0
+BuildRequires: %{?scl_prefix}rubygem(qpid_messaging)
+BuildRequires: %{?scl_prefix_ror}rubygem(rails)
+BuildRequires: %{?scl_prefix_ruby}rubygem(json)
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildRequires: %{?scl_prefix_ruby}ruby(rubygems)
+BuildRequires: %{?scl_prefix_ruby}ruby(release)
+
+#old dependencies that are no longer needed, remove in katello 2.6
+Obsoletes: ruby193-rubygem-haml
+Obsoletes: ruby193-rubygem-haml-rails
+Obsoletes: ruby193-rubygem-hpricot
+Obsoletes: ruby193-rubygem-strong_parameters
+Obsoletes: ruby193-rubygem-tire
+Obsoletes: ruby193-rubygem-less
+Obsoletes: ruby193-rubygem-hashr
+Obsoletes: ruby193-rubygem-foreman_gutterball
+
+BuildArch: noarch
+Provides: %{?scl_prefix}rubygem(katello) = %{version}
+%{?scl:Obsoletes: ruby193-rubygem-%{gem_name}}
+
+%description
+Katello
+
+%package doc
+BuildArch:  noarch
+Requires:   %{?scl_prefix}%{pkg_name} = %{version}-%{release}
+Summary:    Documentation for rubygem-%{gem_name}
+%{?scl:Obsoletes: ruby193-rubygem-%{gem_name}-doc}
+
+%description doc
+This package contains documentation for rubygem-%{gem_name}.
+
+%package -n %{katello_ostree}
+Requires:   %{?scl_prefix}%{pkg_name} = %{version}-%{release}
+Summary:    Katello Ostree Plugin
+
+%description -n %{katello_ostree}
+This package provides the ostree plugin for rubygem-%{gem_name}.
+
+%prep
+%setup -n %{pkg_name}-%{version} -q -c -T
+%{?scl:scl enable %{scl} - <<EOF}
+%gem_install -n %{SOURCE0}
+%{?scl:EOF}
+
+%build
+
+%install
+mkdir -p %{buildroot}%{gem_dir}
+cp -a .%{gem_dir}/* \
+        %{buildroot}%{gem_dir}/
+
+%foreman_bundlerd_file
+%foreman_precompile_plugin -s -a
+
+%files
+%dir %{gem_instdir}/
+%{gem_libdir}
+%{gem_instdir}/app
+%{gem_instdir}/config
+%{gem_instdir}/db
+%{gem_instdir}/locale
+%{gem_instdir}/engines
+%{gem_instdir}/ca
+%{gem_instdir}/vendor
+%{gem_spec}
+%{foreman_bundlerd_plugin}
+%{foreman_apipie_cache_foreman}
+%{foreman_apipie_cache_plugin}
+%{foreman_assets_plugin}
+%{gem_instdir}/public/assets/bastion_katello
+
+%exclude %{gem_cache}
+%exclude %{gem_instdir}/lib/katello/repository_types/ostree.rb
+
+%doc %{gem_instdir}/LICENSE.txt
+
+%files doc
+%doc %{gem_docdir}
+%doc %{gem_instdir}/README.md
+
+%files -n %{katello_ostree}
```

Comment:
If this package is just a single file, why is it split from the main package?

Reply:
Because that single files "turns on" ostree support which is configurable. Until the mainline code is refactored into a better way to handle this, this is how we solved the problem. You could make the argument ostree should not be optional now that we are pure EL7 but that's not entirely up to packaging's per view.

Reply:
So mainline code has no configuration switch for it so it's solved in packaging. Is there an issue tracking this?

Reply:
To be fair, it's a switch in both places. This RPM lays down the configuration. The true answer lies in a repository re-factoring that will be undertaken in the coming months from what I understand.